nautobot/docs/development/apps/api/views/nautobotuiviewset.md

```diff
@@ -145,3 +145,44 @@ For example, for a DetailView template for `YourAppModel`, the template name wil
 If you do not provide your own templates in the `yourapp/templates/yourapp` folder, `NautobotUIViewSet` will fall back to `generic/object_{self.action}.html`.
 
 Since in many cases the `create` and `update` templates for a model will be identical, you are not required to create both. If you provide a `{app_label}/{model_opts.model_name}_create.html` file but not a `{app_label}/{model_opts.model_name}_update.html` file, then when you update an object, it will fall back to `{app_label}/{model_opts.model_name}_create.html` and vice versa.
+
+### Adding Custom Views To NautobotUIViewSet & NautobotUIViewSetRouter
+
++++ 1.6.0
+    Via [PR #4045](https://github.com/nautobot/nautobot/pull/4045), notes and changelog views provided by mixins have now been moved to this pattern.
+
+DjangoRestFramework provides the ability to decorate a method on a ViewSet with `@action(detail=True)` to add the method as a view to the ViewSetRouter. This method must return a fully rendered HTML view.
+
+Below is an example of adding a custom view to the plugin ViewSet, a few considerations to keep in mind is the method name is the `action` and will be used for [template lookup](#template-naming-for-nautobotuiviewset) and URL naming. The expected template must be named `yourapp/yourappmodel_customview.html` and the reversible URL name will be `plugins:yourapp:yourappmodel_customview`.
+
+```python
+from nautobot.apps.views import NautobotUIViewSet
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from yourapp import filters, forms, models, tables
+from yourapp.api import serializers
+
+class YourAppModelUIViewSet(NautobotUIViewSet):
+    bulk_create_form_class = forms.YourAppModelCSVForm
+    bulk_update_form_class = forms.YourAppModelBulkEditForm
+    filterset_class = filters.YourAppModelFilterSet
+    filterset_form_class = forms.YourAppModelFilterForm
+    form_class = forms.YourAppModelForm
+    queryset = models.YourAppModel.objects.all()
+    serializer_class = serializers.YourAppModelSerializer
+    table_class = tables.YourAppModelTable
+    custom_action_permission_map = {"customview": "my_cutom_action"}
+    # custom_action_permission_map = {"customview": "view"}  # Can also use built in actions
+
+    @action(detail=True)
+    def customview(self, request, *args, **kwargs):
+        """Context passed to template for rendering.
+        
+        Expected URL pattern will be `/plugins/yourapp/yourappmodel/<uuid>/customview/`
+        """
+        context = {
+            "some_key": "some value",
+        }
+        return Response(context)
+```
```

Comment:
Therefore all of this wouldn't be necessary, because all you'd have to do is define the custom action and it would "just work™".